add Coprocessor cache doc (#2616)
pingcap/docs: coprocessor-cache.md

---
title: Coprocessor Cache
summary: Learn the features of Coprocessor Cache.
category: reference
---

# Coprocessor Cache

Starting from v4.0, the TiDB instance supports caching the results of the calculation that is pushed down to TiKV (the Coprocessor Cache feature), which can accelerate the calculation process in some scenarios.

## Configuration

You can configure Coprocessor Cache via the `tikv-client.copr-cache` configuration items in the TiDB configuration file. For details about how to enable and configure Coprocessor Cache, see [TiDB Configuration File](/tidb-configuration-file.md#tikv-clientcopr-cache-new-in-v400).

## Feature description

+ When a SQL statement is executed on a single TiDB instance for the first time, the execution result is not cached.

> + Calculation results are cache in the memory of TiDB. If the TiDB instance is restarted, the cache becomes invalid.

Reviewer: ```suggestion
+ Calculation results are cached in the memory of TiDB. If the TiDB instance is restarted, the cache becomes invalid.
```

+ The cache is not shared among TiDB instances.

> + The cached result is the result of push-down calculations. Even if the result hits the cache, TiDB still performs calculation.

Reviewer: ```suggestion
+ Only push-down calculation result is cached. Even if cache is hit, TiDB still need to perform subsequent calculation.
```
